### ISSUE TITLE
PHAssetCollection .count triggers empty_count rule

### ISSUE THREAD

**Issue by elomonaco** [realm/SwiftLint#4601]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

when I run the lint I get this error:

> error: Empty Count Violation: Prefer checking `isEmpty` over comparing `count` to zero. (empty_count)

however the code in question is this:

```swift
let assets = PHAsset.fetchAssets(in: self, options: nil)

return assets.count > 0
```

as far as I can tell, PHAssetCollection doesn't have the isEmpty helper attached to it so using .count is a valid check in this case.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file:

```yml
disabled_rules: # rule identifiers turned on by default to exclude from running
  - class_delegate_protocol
  - cyclomatic_complexity
  - file_length
  - function_body_length
  - inclusive_language
  - line_length
  - switch_case_alignment
  - todo
  - type_body_length
opt_in_rules: # some rules are turned off by default, so you need to opt-in
  - empty_count # Find all the available rules by running: `swiftlint rules`

# Alternatively, specify all rules explicitly by uncommenting this option:
# only_rules: # delete `disabled_rules` & `opt_in_rules` if using this
#   - empty_parameters
#   - vertical_whitespace

analyzer_rules: # Rules run by `swiftlint analyze`
  - explicit_self

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
force_cast: warning # implicitly
force_try:
  severity: warning # explicitly
# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
  min_length: 2 # only warning
  max_length: # warning and error
    warning: 70
    error: 90
  excluded: iPhone # excluded via string
  allowed_symbols: ["_"] # these are allowed in type names
identifier_name:
  min_length: 1
  max_length: # warning and error
    warning: 70
    error: 90
reporter: "xcode" # reporter type (xcode, json, csv, checkstyle, codeclimate, junit, html, emoji, sonarqube, markdown, github-actions-logging)
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. no
* Which Xcode version are you using (check `xcodebuild -version`)? 14.1 Build version 14B47b
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
import UIKit
import Photos

extension PHAssetCollection {
    func hasAssets() -> Bool {
        let assets = PHAsset.fetchAssets(in: self, options: nil)

        return assets.count > 0
    }
}
```

**Comment by jpsim** [comment]
This is by design. SwiftLint operates on the pre-typechecked AST, so it has no way of knowing what type `assets` is. You can disable the rule locally here with a `// swiftlint:disable ...` command, or disable it in your `.swiftlint.yml` config.

**Comment by elomonaco** [comment]
I for now just set the severity to warning so the lint didn't fail the build, I was aware of that comment tag so I'll use that. Thanks
